src/text/encoder.rs

```diff
@@ -0,0 +1,264 @@
+use crate::result::IonResult;
+use crate::IonType;
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, FixedOffset};
+use std::io;
+
+/// Utility type for writing scalar Rust values as Ion text.
+pub struct TextEncoder {}
+
+impl TextEncoder {
+    #[inline(always)]
+    fn null_text(ion_type: IonType) -> &'static str {
+        use IonType::*;
+        match ion_type {
+            Null => "null",
+            Boolean => "null.bool",
+            Integer => "null.int",
+            Float => "null.float",
+            Decimal => "null.decimal",
+            Timestamp => "null.timestamp",
+            Symbol => "null.symbol",
+            String => "null.string",
+            Blob => "null.blob",
+            Clob => "null.clob",
+            List => "null.list",
+            SExpression => "null.sexp",
+            Struct => "null.struct",
+        }
+    }
+
+    #[inline(always)]
+    fn bool_text(value: bool) -> &'static str {
+        match value {
+            true => "true",
+            false => "false",
+        }
+    }
+
+    /// Writes a null of the specified Ion type to `output`.
+    pub fn encode_null<W: io::Write>(output: &mut W, ion_type: IonType) -> IonResult<()> {
+        write!(output, "{}", TextEncoder::null_text(ion_type))?;
+        Ok(())
+    }
+
+    /// Writes a bool value to `output` as an Ion boolean.
+    pub fn encode_bool<W: io::Write>(output: &mut W, value: bool) -> IonResult<()> {
+        write!(output, "{}", TextEncoder::bool_text(value))?;
+        Ok(())
+    }
+
+    /// Writes an i64 value to `output` as an Ion integer.
+    pub fn encode_i64<W: io::Write>(output: &mut W, value: i64) -> IonResult<()> {
+        write!(output, "{}", value)?;
+        Ok(())
+    }
+
+    /// Writes a BigDecimal value to `output` as an Ion decimal.
+    pub fn encode_big_decimal<W: io::Write>(output: &mut W, value: &BigDecimal) -> IonResult<()> {
+        write!(output, "{}", &value)?;
+        Ok(())
+    }
```

Comment:
I considered this. The trade-off is that you replace `output: &mut W` everywhere with `&mut self` and then callers need to create an instance of `TextEncoder` anywhere they want to use it. Having functions instead of methods felt like a minor ergonomics win.

I have a `TODO` for creating a reusable `String` for formatting `blob` values; having `TextEncoder` own the `writer` *and* a `String` buffer seems like enough value to tip things in the other direction. I'll go ahead and make that change.

Reply:
> Having functions instead of methods felt like a minor ergonomics win.

What is the ergonomic win? The call sites seem roughly the same to me (except the writer is a parameter vs. bound to the method receiver) so it seems like a wash, but I could be missing something.